pkg/cloudfoundry/ManifestUtils.go

```diff
@@ -0,0 +1,226 @@
+package cloudfoundry
+
+import (
+	"fmt"
+	"github.com/ghodss/yaml"
+	"github.com/pkg/errors"
+	"io/ioutil"
+	"reflect"
+
+	"github.com/SAP/jenkins-library/pkg/log"
+)
+
+const propApplications = "applications"
+const propBuildpacks = "buildpacks"
+const propBuildpack = "buildpack"
+
+// Manifest ...
+type Manifest struct {
+	self     map[string]interface{}
+	modified bool
+	name     string
+}
+
+var m Manifest
+
+var _readFile = ioutil.ReadFile
+
+// ReadManifest ...
+func ReadManifest(name string) (Manifest, error) {
+
+	log.Entry().Infof("Reading manifest file  '%s'", name)
+
+	m := Manifest{self: make(map[string]interface{}), name: name, modified: false}
+
+	content, err := _readFile(name)
+	if err != nil {
+		return m, errors.Wrapf(err, "cannot read file '%v'", m.name)
```

Comment:
In case we return `nil` the compiler hits us with:

"`cannot use nil as type Manifest in return argument`"

With the current - compiling - approach we return in fact some empty manifest.

To my understanding in go it is the duty of the caller to apply a proper check for the error. In case of an error any created instance must not be used later on.

So far I do not properly understand the advantages of exception handling in go (... returning by convention any error simply as the last return value), but that is another topic ...